rotkehlchen/db/custom_assets.py

```diff
@@ -0,0 +1,111 @@
+        the query and the count of custom assets in the DB.
+        """
+        entries = self._get_custom_assets(filter_query=filter_query)
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            query, bindings = filter_query.prepare(with_pagination=False)
+            query = 'SELECT COUNT(*) from custom_assets AS A JOIN assets AS B ON A.identifier = B.identifier ' + query  # noqa: E501
+            entries_found = cursor.execute(query, bindings).fetchone()[0]
+            entries_total = cursor.execute('SELECT COUNT(*) FROM custom_assets;').fetchone()[0]
+            return entries, entries_found, entries_total
+
+    def add_custom_asset(self, custom_asset: CustomAsset) -> str:
+        """
+        Adds a custom new custom asset into `assets` and `custom_assets` table.
+        May raise:
+        - InputError if the combination of the name & type / asset with identifier already exist.
+        """
+        self.check_if_custom_asset_exists(custom_asset)
+        with GlobalDBHandler().conn.write_ctx() as global_db_write_cursor, self.db.user_write() as db_write_cursor:  # noqa: E501
+            global_db_write_cursor.execute(
+                'INSERT INTO assets(identifier, name, type) VALUES (?, ?, ?)',  # noqa: E501
+                (
+                    custom_asset.identifier,
+                    custom_asset.name,
+                    AssetType.CUSTOM_ASSET.serialize_for_db(),
+                ),
+            )
+            global_db_write_cursor.execute(
+                'INSERT INTO custom_assets(identifier, type, notes) VALUES(?, ?, ?)',
+                custom_asset.serialize_for_db(),
+            )
+            self.db.add_asset_identifiers(db_write_cursor, [custom_asset.identifier])
+        return custom_asset.identifier
+
+    def edit_custom_asset(self, custom_asset: CustomAsset) -> None:
+        """
+        Edits a custom asset.
+        May raise:
+        - InputError if editing a custom asset that does not exist or combination
+        of name and type is already present.
+        """
+        self.check_if_custom_asset_exists(custom_asset)
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            write_cursor.execute(
+                'UPDATE assets SET name=? WHERE identifier=?',
+                (custom_asset.name, custom_asset.identifier),
+            )
+            write_cursor.execute(
+                'UPDATE custom_assets SET notes=?, type=? WHERE identifier=?',
+                (custom_asset.notes, custom_asset.custom_asset_type, custom_asset.identifier),
+            )
+            if write_cursor.rowcount == 0:
+                raise InputError(
+                    f'Tried to edit custom asset with identifier {custom_asset.identifier} and name '  # noqa: E501
+                    f'{custom_asset.name} but it was not found',
+                )
+
+    @staticmethod
+    def check_if_custom_asset_exists(custom_asset: CustomAsset) -> None:
```

Comment:
1. This is a private method only in this module so start with `_`
2. You don't check since this does not return `bool`. You raise.
```suggestion
    def _raise_if_custom_asset_exists(custom_asset: CustomAsset) -> None:
```